Documentation/proposals/oci.md

```diff
@@ -0,0 +1,127 @@
+
+### Goals, non-Goals
+
+With the deprecation of the appc Spec (https://github.com/appc/spec#-disclaimer-) the current internal rkt architecture is not favorable any more.
+Currently rkt does support ACI, Docker, and OCI images, but the conversion step from OCI to ACI using `docker2aci` seems unnecessary.
+It introduces CPU and I/O bound overhead and is bound by semantical differences between the formats. For these reasons native support of OCI images inside rkt is envisioned.
+
+The goal therefore is to support OCI images natively next to ACI.
+
+This document outlines the following necessary steps and references existing work to transition to native OCI image support in rkt:
+
+1. Distribution points
+2. Reference based image handling
+3. Transport handlers
+4. Tree store support for OCI
+
+A non-goal is the implementation of the [OCI runtime specification](https://github.com/opencontainers/runtime-spec). There is ongoing work in [https://github.com/coreos/rkt/issues/3408](https://github.com/coreos/rkt/issues/3408) covering this topic.
+
+### Overview
+
+#### Distribution points
+
+rkt historically used the image name and heuristics around it to determine the actual image format type (appc, Docker, OCI).
+The concept of "distribution points" introduced a URI syntax that uniquely points to the different image formats including the necessary metadata (file location, origin URL, version, etc.).
+
+The URI scheme "cimd" (Container Image Distribution) was chosen to uniquely identify different image formats. The following CIMDs are currently supported:
+
+Name | Example
+-----|--------
+appc | `cimd:appc:v=0:coreos.com/etcd?version=v3.0.3&os=linux&arch=amd64`
+ACIArchive | `cimd:aci-archive:v=0:file%3A%2F%2Fabsolute%2Fpath%2Fto%2Ffile`
+Docker | `cimd:docker:v=0:busybox`
+
+The design document can be found in https://github.com/coreos/rkt/blob/master/Documentation/devel/distribution-point.md.
+
+###### Status
+
+- The design document (https://github.com/coreos/rkt/pull/2953) is merged.
+- The implementation (https://github.com/coreos/rkt/pull/3369) is merged.
+
+###### TODOs
+
+- Introduce a dedicated remote `cimd:oci` and potentially also a local `cimd:oci-layout` (see [github.com/opencontainers/image-spec/image-layout.md](https://github.com/opencontainers/image-spec/blob/master/image-layout.md)) CIMD scheme.
+
+#### Reference based image handling
+
+The current image store implementation does not support different image formats. The blob image store only supports SHA-512.
+The ql backed SQL image store has a simple SQL scheme referencing only ACI images.
+
+In order to prepare native support for OCI the following changes need to be implemented:
+
+- Store the CIMD URI as a primary key in the current image store.
+- Support for multiple hash algorithms: Currently only SHA-512 is supported. OCI in addition needs SHA-256 and optionally SHA-384 and SHA-512.
```

Comment:
I saw it in this proposal https://github.com/coreos/rkt/blob/23313af1c3dac2fb24fe41f9a7c5eaca573e45dd/Documentation/proposals/reference_based_image_access_and_cas_store.md#reference-based-image-access-on-a-cas-for-both-appc-aci-and-oci-images-proposal (last paragraph).

But you are right, I'll rephrase this sentence it since there is no mention of it in the spec itself.